guides/upgrading/0.2.x_to_0.3.x.md

```diff
@@ -0,0 +1,113 @@
+# Upgrading from 0.2.x to 0.3.x
+
+Version `0.3.0` dropped support for some Elasticsearch 5.x features that have
+been deprecated or removed in Elasticsearch 6.x.
+
+## Rationale
+
+As noted in the Elasticsearch document ["Removal of mapping
+types"](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/removal-of-types.html),
+Elasticsearch no longer supports multiple document types per index.
+
+Instead, each index is expected to have a single type, `_doc`, and handle
+parent/child relationships with the new
+[join](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/parent-join.html)
+field.
+
+To support this, version `0.3.0` of this client library makes some breaking
```

Comment:
just curious @danielberkompas: my understanding of semver is that breaking changes require a new major version, no? otherwise you're (as a user) going with a minor update (`0.2.0` to `0.3.0`) and for sure not expecting a breaking changes to be happening.

Reply:
ah yes, i realize the same would have been already the case for our changes from `0.1.1` to `0.2.0` - which probably is the reason my patches were all based on the former, because it's no simple upgrade.

Reply:
Semver rules regarding major versions and breaking changes only kick in after you have reached 1.0.0. Anything < 1.0.0 is considered a beta version which can include breaking changes. 

> 4. Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

— [Semver.org](https://semver.org/#spec-item-4)

Only after you reach 1.0.0 are you required to have a commitment to stability on major versions.

Reply:
again something learned today, thanks for the pointer!